src/runtime_src/core/common/device.cpp

```diff
@@ -78,6 +78,53 @@ register_axlf(const axlf* top)
 
   // Build modified CONNECTIVITY and MEM_TOPOLOGY section based on memory group ids
   // Base groups off data from driver
+  auto m_itr = m_axlf_sections.find(MEM_TOPOLOGY);
+  int g_count = (int)m_grp_info.size();
+  if (g_count && (m_itr != m_axlf_sections.end())) {
+    auto m_mem = reinterpret_cast<::mem_topology*>((*m_itr).second.data());
+    
+    /* Store the original memory topology before modify */
+    size_t size = (*m_itr).second.size();
+    char *buf = new char[size];
+    memcpy(buf, m_mem, size);
+    
+    auto m_topo = reinterpret_cast<::mem_topology*>(buf);
+    for (auto i=0; i<g_count; ++i) {
+      auto& mem = m_mem->m_mem_data[i];
+      auto it = m_grp_info.find(i);
+      if (it != m_grp_info.end()) {
+        auto l_idx = (*it).second.first;
+        auto h_idx = (*it).second.second;
+	mem.m_base_address = m_topo->m_mem_data[l_idx].m_base_address;
+        mem.m_size = (m_topo->m_mem_data[h_idx].m_base_address + m_topo->m_mem_data[h_idx].m_size) -
+			m_topo->m_mem_data[l_idx].m_base_address;
+        mem.m_type = m_topo->m_mem_data[l_idx].m_type;
+        memcpy(&mem.m_tag, m_topo->m_mem_data[l_idx].m_tag, sizeof(m_topo->m_mem_data[l_idx].m_tag));
+        mem.m_used = 1;
+      }
+    }
+    // Update remaing entries as un-used
+    for (auto i=g_count; i<m_mem->m_count; ++i) {
+	    auto& mem = m_mem->m_mem_data[i];
+	    mem.m_used = 0;
+    }
+    // Update the new memory topology count 
+    m_mem->m_count = g_count;
+    delete []buf;
+  }
+
+  auto c_itr = m_axlf_sections.find(CONNECTIVITY);
+  if (c_itr != m_axlf_sections.end()) {
+    auto m_con = reinterpret_cast<::connectivity*>((*c_itr).second.data());
```

Comment:
Maybe use the get_axlf_section templated membor function?

Reply:
The return type of get_axlf_section if std::pair<const char*, size_t>. But in our case, we need to modify the connectivity, which is not allowed because of const.

Reply:
yep, see other response.